improver_tests/threshold/test_BasicThreshold.py

```diff
@@ -487,31 +503,92 @@ def test_threshold_point_nan(self):
         # Need to copy the cube as we're adjusting the data.
         self.cube.data[2][2] = np.NAN
         msg = "NaN detected in input cube data"
-        plugin = Threshold(2.0, fuzzy_factor=self.fuzzy_factor, comparison_operator="<")
-        with self.assertRaisesRegex(ValueError, msg):
-            plugin(self.cube)
-
-    def test_each_threshold_func(self):
-        """Test user supplied func is applied on each threshold cube."""
-        # Need to copy the cube as we're adjusting the data.
-        new_attr = {"new_attribute": "narwhal"}
         plugin = Threshold(
-            2.0,
-            each_threshold_func=lambda cube: cube.attributes.update(new_attr) or cube,
+            threshold_values=2.0,
+            fuzzy_factor=self.fuzzy_factor,
+            comparison_operator="<",
         )
-        result = plugin(self.cube)
-        self.assertTrue("new_attribute" in result.attributes)
+        with self.assertRaisesRegex(ValueError, msg):
+            plugin(self.cube)
 
     def test_cell_method_updates(self):
         """Test plugin adds correct information to cell methods"""
         self.cube.add_cell_method(CellMethod("max", coords="time"))
-        plugin = Threshold(2.0, comparison_operator=">")
+        plugin = Threshold(threshold_values=2.0, comparison_operator=">")
         result = plugin(self.cube)
         (cell_method,) = result.cell_methods
         self.assertEqual(cell_method.method, "max")
         self.assertEqual(cell_method.coord_names, ("time",))
         self.assertEqual(cell_method.comments, ("of precipitation_amount",))
 
+    def test_threshold_vicinity(self):
```

Comment:
We need vicinity tests with a land-sea cube too, and a test for a land-sea cube without vicinity processing (which should raise an error).

I also think most tests need to run with and without a realization coordinate, and with a scalar realization coordinate too.